packages/flutter/test/widgets/table_test.dart

```diff
@@ -376,6 +376,32 @@ void main() {
     // If the above bug is present this test will never terminate.
   });
 
+    testWidgets('Really small deficit double precision error (redistributing deficit)', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/36256
+    const SizedBox cell = SizedBox(width: 1, height: 1);
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Table(
+          columnWidths: const <int, TableColumnWidth>{
+            0: FlexColumnWidth(1.0),
+            1: FlexColumnWidth(0.123),
+            2: FlexColumnWidth(0.123),
+            3: FlexColumnWidth(0.123),
+            4: FlexColumnWidth(0.123),
+            5: FlexColumnWidth(0.123),
+            6: FlexColumnWidth(0.123),
+          },
+          children: <TableRow>[
+            TableRow(children: List<Widget>.filled(7, cell)),
+            TableRow(children: List<Widget>.filled(7, cell)),
+          ],
+        ),
+      ),
+    );
+    // If the above bug is present this test will never terminate.
```

Comment:
Is there a way we can make this test pass/fail instead of pass/hang?

Reply:
I've tried some approaches, but to no avail, since the main thread is stuck in an endless loop, it never allows interruptions to finish the test early. I thought running `pumpWidget` in a separate isolate (via `compute`) would work, but I got the following error when attempting that (seems to be partial?):
```
../../third_party/dart/runtime/vm/raw_object_snapshot.cc: 523: error: unreachable code
Shell: version=2.5.0-dev.1.0.flutter-24725a8559 (Thu Jul 11 11:40:03 2019 +0000) on "linux_x64"
Shell: thread=20327, isolate=main(0x5023370)
Shell:   pc 0x0000000002e8e92c fp 0x00007ffd578277b0
/home/kps/Workspace/flutter/flutter/bin/cache/artifacts/engine/linux-x64/flutter_tester+0x2c8e92c
Shell: -- End of DumpStackTrace
```

Reply:
(It's probably me misusing the `compute` method, though)

Reply:
Perhaps instead of trying to stop an infinite loop in the test, you could assert in the computeColumnWidths method that the deficit or minimumDeficit is never a value that would lead to this hanging process?

Reply:
I noticed that the fix for the similar issue you referenced (#27083) utilized a similar test (#27112), so this may actually be ok. 
cc/ @jonahwilliams

Reply:
You can't invoke dart UI methods from a separate isolate unfortunately.

The problem is that it becomes very hard to tell the difference between a test that never terminates and a test that's just running more slowly and timing out.  That said, I can't think of a betterw ay to test this.  Maybe we can just move this comment up to the top of the test method and clarify that a timeout on this test may indicate a regression on a never-terminating loop in the framework.

Reply:
@kpsroka do you think it would make more sense to test against the rendering layer e.g. `computeMinIntrinsicHeight` and `performLayout`?  And we should probably name the test something like `Computing small widths can terminate`.  That should make it a bit clearer what we're testing and where the bug is.

Reply:
(that test would be in `test/rendering/table_test.dart`)

I still wouldn't use an isolate to check it, because that will be prone to flakiness as much as just letting the test timeout.

Reply:
@dnfield I've now added a similar test in `rendering/table_test.dart`. Please let me know which one seems more adequate to you. The one in `rendering` is obviously doing less work, as it just triggers layout(), but readability-wise they are very similar.

For the timeout, I introduced `assert(delta != 0)` in the while loop, that should fail when the erroneous condition happens again.

As for the test name, it's not necessarily 'small widths' that are the problem, but minuscule difference between calculated flex width and constraints. I'll think about a better name.

Reply:
Having both is good - if they fail it will be a little more obvious about what is causing the problem now.